File: app/src/main/java/com/wa/pranksound/ui/component/activity/MainActivity.java
package com.wa.pranksound.ui.component.activity;

import androidx.activity.OnBackPressedCallback;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.FragmentManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.Dialog;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.wa.pranksound.R;
import com.wa.pranksound.ui.component.fragment.CreateFragment;
import com.wa.pranksound.ui.component.fragment.FavoritesFragment;
import com.wa.pranksound.ui.component.fragment.HomeFragment;
import com.wa.pranksound.ui.component.fragment.LeaderBoardFragment;
import com.wa.pranksound.ui.component.settings.SettingsActivity;
import com.wa.pranksound.utils.BaseActivity;
import com.wa.pranksound.utils.Gdpr;

import java.util.Objects;

public class MainActivity extends BaseActivity {
    ImageView imgHome, imgCreate, imgFavorites, imgLeaderboard;
    RecyclerView rcv_cate;
    FragmentManager fragmentManager;

    LinearLayout ll_home, ll_create, ll_favorites, ll_leaderboard;
    String navTab = "home";
    TextView tv_title, tv_home, tv_create, tv_favorites, tv_leaderboard;
    ImageButton btnSettings;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findView();

        fragmentManager = getSupportFragmentManager();
        bottomNavigationClick();
        String get_record = getIntent().getStringExtra("from_record");

        if (get_record != null) {
            changeFragment("create");
        } else {
            changeFragment("home");
        }

        btnSettings.setOnClickListener(v -> {
            startActivity(new Intent(MainActivity.this, SettingsActivity.class));
        });
    }

    private void changeFragment(String tab) {
        switch (tab) {
            case "home":
                fragmentManager.beginTransaction().replace(R.id.fragmentMain, new HomeFragment()).commit();
                tv_title.setText(R.string.home);

                imgHome.setBackgroundResource(R.drawable.ic_home_selected);
                tv_home.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color_selected));

                imgCreate.setBackgroundResource(R.drawable.ic_create_unselected);
                tv_create.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgFavorites.setBackgroundResource(R.drawable.ic_fav_unselected);
                tv_favorites.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_unselected);
                tv_leaderboard.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                break;

            case "create":
                fragmentManager.beginTransaction().replace(R.id.fragmentMain, new CreateFragment()).commit();
                tv_title.setText(R.string.create_sound);

                imgHome.setBackgroundResource(R.drawable.ic_home_unselected);
                tv_home.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgCreate.setBackgroundResource(R.drawable.ic_create_selected);
                tv_create.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color_selected));

                imgFavorites.setBackgroundResource(R.drawable.ic_fav_unselected);
                tv_favorites.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_unselected);
                tv_leaderboard.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));
                break;

            case "favorites":
                fragmentManager.beginTransaction().replace(R.id.fragmentMain, new FavoritesFragment()).commit();
                tv_title.setText(R.string.favorites);

                imgHome.setBackgroundResource(R.drawable.ic_home_unselected);
                tv_home.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgCreate.setBackgroundResource(R.drawable.ic_create_unselected);
                tv_create.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgFavorites.setBackgroundResource(R.drawable.ic_fav_selected);
                tv_favorites.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color_selected));

                imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_unselected);
                tv_leaderboard.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));
                break;

            case "leaderboard":
                fragmentManager.beginTransaction().replace(R.id.fragmentMain, new LeaderBoardFragment()).commit();
                tv_title.setText(R.string.leaderboard);

                imgHome.setBackgroundResource(R.drawable.ic_home_unselected);
                tv_home.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgCreate.setBackgroundResource(R.drawable.ic_create_unselected);
                tv_create.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgFavorites.setBackgroundResource(R.drawable.ic_fav_unselected);
                tv_favorites.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_selected);
                tv_leaderboard.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color_selected));
                break;
            default:
                fragmentManager.beginTransaction().replace(R.id.fragmentMain, new HomeFragment()).commit();
                tv_title.setText(R.string.home);

                imgHome.setBackgroundResource(R.drawable.ic_home_selected);
                tv_home.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color_selected));

                imgCreate.setBackgroundResource(R.drawable.ic_create_unselected);
                tv_create.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgFavorites.setBackgroundResource(R.drawable.ic_fav_unselected);
                tv_favorites.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));

                imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_unselected);
                tv_leaderboard.setTextColor(ContextCompat.getColor(this, R.color.bottom_nav_text_color));
                break;
        }
    }

    private void bottomNavigationClick() {
        ll_home.setOnClickListener(v -> {
            navTab = "home";
            changeFragment(navTab);
        });

        ll_create.setOnClickListener(v -> {
            navTab = "create";
            changeFragment(navTab);
        });

        ll_favorites.setOnClickListener(v -> {
            navTab = "favorites";
            changeFragment(navTab);
        });

        ll_leaderboard.setOnClickListener(v -> {
            navTab = "leaderboard";
            changeFragment(navTab);
        });
    }

    private void findView() {

        rcv_cate = findViewById(R.id.rcv_cate);

        ll_home = findViewById(R.id.ll_home);
        ll_create = findViewById(R.id.ll_create);
        ll_favorites = findViewById(R.id.ll_favorites);
        ll_leaderboard = findViewById(R.id.ll_leaderboard);

        tv_title = findViewById(R.id.tv_title);

        imgHome = findViewById(R.id.img_home);
        imgCreate = findViewById(R.id.img_create);
        imgFavorites = findViewById(R.id.img_favorites);
        imgLeaderboard = findViewById(R.id.img_leaderboard);

        tv_home = findViewById(R.id.tv_home);
        tv_create = findViewById(R.id.tv_create);
        tv_favorites = findViewById(R.id.tv_favorites);
        tv_leaderboard = findViewById(R.id.tv_leaderboard);
        btnSettings = findViewById(R.id.btnSettings);

        new Gdpr().make(this);
        getOnBackPressedDispatcher().addCallback(new OnBackPressedCallback(true) {
            @Override
            public void handleOnBackPressed() {
                showExitDialog();
            }
        });
    }

    public void showExitDialog() {
        Dialog dialogCustomExit = new Dialog(MainActivity.this);
        dialogCustomExit.requestWindowFeature(Window.FEATURE_NO_TITLE);
        Objects.requireNonNull(dialogCustomExit.getWindow()).setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        WindowManager.LayoutParams lp = new WindowManager.LayoutParams();
        lp.copyFrom(dialogCustomExit.getWindow().getAttributes());
        lp.width = WindowManager.LayoutParams.MATCH_PARENT;
        lp.height = WindowManager.LayoutParams.MATCH_PARENT;
        dialogCustomExit.setContentView(R.layout.exit_dialog_layout);
        dialogCustomExit.setCancelable(true);
        dialogCustomExit.show();
        dialogCustomExit.getWindow().setAttributes(lp);

        TextView btnNegative = dialogCustomExit.findViewById(R.id.btnNegative);
        TextView btnPositive = dialogCustomExit.findViewById(R.id.btnPositive);

        btnPositive.setOnClickListener(v -> {
            dialogCustomExit.dismiss();
            finishAffinity();

        });
        btnNegative.setOnClickListener(v -> dialogCustomExit.dismiss());
    }
}

File: app/src/main/java/com/joya/pranksound/ui/component/main/MainActivity.kt
package com.joya.pranksound.ui.component.main

import android.app.Dialog
import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.Window
import android.view.WindowManager
import android.widget.TextView
import androidx.activity.OnBackPressedCallback
import androidx.core.content.ContextCompat
import androidx.fragment.app.FragmentManager
import com.joya.pranksound.R
import com.joya.pranksound.databinding.ActivityMainBinding
import com.joya.pranksound.ui.base.BaseBindingActivity
import com.joya.pranksound.ui.component.fragment.CreateFragment
import com.joya.pranksound.ui.component.fragment.FavoritesFragment
import com.joya.pranksound.ui.component.fragment.HomeFragment
import com.joya.pranksound.ui.component.fragment.LeaderBoardFragment
import com.joya.pranksound.ui.component.settings.SettingsActivity
import com.joya.pranksound.utils.Gdpr
import com.joya.pranksound.utils.extention.setOnSafeClick

class MainActivity : BaseBindingActivity<ActivityMainBinding, MainViewModel>() {
    private var fragmentManager: FragmentManager? = null
    private var navTab: String = "home"

    override val layoutId: Int
        get() = R.layout.activity_main

    override fun getViewModel(): Class<MainViewModel> = MainViewModel::class.java

    override fun setupData() {
    }

    override fun setupView(savedInstanceState: Bundle?) {
        findView()

        fragmentManager = supportFragmentManager
        bottomNavigationClick()
        val getRecord = intent.getStringExtra("from_record")

        if (getRecord != null) {
            changeFragment("create")
        } else {
            changeFragment("home")
        }

        binding.btnSettings.setOnClickListener {
            startActivity(
                Intent(
                    this@MainActivity,
                    SettingsActivity::class.java
                )
            )
        }


    }

    private fun changeFragment(tab: String) {
        when (tab) {
            "home" -> {
                fragmentManager!!.beginTransaction().replace(R.id.fragmentMain, HomeFragment())
                    .commit()
                binding.tvTitle.setText(R.string.home)

                binding.imgHome.setBackgroundResource(R.drawable.ic_home_selected)
                binding.tvHome.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color_selected
                    )
                )

                binding.imgCreate.setBackgroundResource(R.drawable.ic_create_unselected)
                binding.tvCreate.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgFavorites.setBackgroundResource(R.drawable.ic_fav_unselected)
                binding.tvFavorites.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_unselected)
                binding.tvLeaderboard.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )
            }

            "create" -> {
                fragmentManager!!.beginTransaction().replace(R.id.fragmentMain, CreateFragment())
                    .commit()
                binding.tvTitle.setText(R.string.create_sound)

                binding.imgHome.setBackgroundResource(R.drawable.ic_home_unselected)
                binding.tvHome.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgCreate.setBackgroundResource(R.drawable.ic_create_selected)
                binding.tvCreate.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color_selected
                    )
                )

                binding.imgFavorites.setBackgroundResource(R.drawable.ic_fav_unselected)
                binding.tvFavorites.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_unselected)
                binding.tvLeaderboard.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )
            }

            "favorites" -> {
                fragmentManager!!.beginTransaction().replace(R.id.fragmentMain, FavoritesFragment())
                    .commit()
                binding.tvTitle.setText(R.string.favorites)

                binding.imgHome.setBackgroundResource(R.drawable.ic_home_unselected)
                binding.tvHome.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgCreate.setBackgroundResource(R.drawable.ic_create_unselected)
                binding.tvCreate.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgFavorites.setBackgroundResource(R.drawable.ic_fav_selected)
                binding.tvFavorites.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color_selected
                    )
                )

                binding.imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_unselected)
                binding.tvLeaderboard.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )
            }

            "leaderboard" -> {
                fragmentManager!!.beginTransaction()
                    .replace(R.id.fragmentMain, LeaderBoardFragment()).commit()
                binding.tvTitle.setText(R.string.leaderboard)

                binding.imgHome.setBackgroundResource(R.drawable.ic_home_unselected)
                binding.tvHome.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgCreate.setBackgroundResource(R.drawable.ic_create_unselected)
                binding.tvCreate.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgFavorites.setBackgroundResource(R.drawable.ic_fav_unselected)
                binding.tvFavorites.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_selected)
                binding.tvLeaderboard.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color_selected
                    )
                )
            }

            else -> {
                fragmentManager!!.beginTransaction().replace(R.id.fragmentMain, HomeFragment())
                    .commit()
                binding.tvTitle.setText(R.string.home)

                binding.imgHome.setBackgroundResource(R.drawable.ic_home_selected)
                binding.tvHome.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color_selected
                    )
                )

                binding.imgCreate.setBackgroundResource(R.drawable.ic_create_unselected)
                binding.tvCreate.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgFavorites.setBackgroundResource(R.drawable.ic_fav_unselected)
                binding.tvFavorites.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )

                binding.imgLeaderboard.setBackgroundResource(R.drawable.ic_leader_board_unselected)
                binding.tvLeaderboard.setTextColor(
                    ContextCompat.getColor(
                        this,
                        R.color.bottom_nav_text_color
                    )
                )
            }
        }
    }

    private fun bottomNavigationClick() {
        binding.apply {
            llHome.setOnSafeClick {
                navTab = "home"
                changeFragment(navTab)
            }

            llCreate.setOnSafeClick {
                navTab = "create"
                changeFragment(navTab)
            }

            llFavorites.setOnSafeClick {
                navTab = "favorites"
                changeFragment(navTab)
            }

            llLeaderboard.setOnSafeClick {
                navTab = "leaderboard"
                changeFragment(navTab)
            }
        }
    }

    private fun findView() {
        Gdpr().make(this)
        onBackPressedDispatcher.addCallback(object : OnBackPressedCallback(true) {
            override fun handleOnBackPressed() {
                showExitDialog()
            }
        })
    }

    fun showExitDialog() {
        val dialogCustomExit = Dialog(this@MainActivity)
        dialogCustomExit.requestWindowFeature(Window.FEATURE_NO_TITLE)
        dialogCustomExit.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        val lp = WindowManager.LayoutParams()
        lp.copyFrom(dialogCustomExit.window!!.attributes)
        lp.width = WindowManager.LayoutParams.MATCH_PARENT
        lp.height = WindowManager.LayoutParams.MATCH_PARENT
        dialogCustomExit.setContentView(R.layout.exit_dialog_layout)
        dialogCustomExit.setCancelable(true)
        dialogCustomExit.show()
        dialogCustomExit.window!!.attributes = lp

        val btnNegative = dialogCustomExit.findViewById<TextView>(R.id.btnNegative)
        val btnPositive = dialogCustomExit.findViewById<TextView>(R.id.btnPositive)

        btnPositive.setOnSafeClick {
            dialogCustomExit.dismiss()
            finishAffinity()
        }
        btnNegative.setOnSafeClick { dialogCustomExit.dismiss() }
    }
}